iceoryx_posh/include/iceoryx_posh/iceoryx_posh_types.hpp

```diff
@@ -148,7 +148,15 @@ constexpr uint32_t MAX_NUMBER_OF_INSTANCES = 50U;
 constexpr uint32_t MAX_RUNNABLE_NUMBER = 1000U;
 constexpr uint32_t MAX_RUNNABLE_PER_PROCESS = 50U;
 
+#if defined(__APPLE__)
+/// @note on macOS the process name length nedds to be decreased since the process name is used for the unix domain
+/// socket path which has a capacity for only 103 characters. The full path consists of UnixDomainSocket::PATH_PREFIX,
+/// which is currently 5 characters and the specified process name.
+constexpr uint32_t MAX_PROCESS_NAME_LENGTH = 98U;
```

Comment:
So the length of PATH_PREFIX is implicitly hidden in the 98. Can you calculate this with the given 103 and PATH_PREFIX. Would be more robust to PATH_PREFIX changes

Reply:
@budrus elfenpiff has plans to make PATH_PREFIX customizable, then this calculation might be obsolete. But could right now be done easily. There are runtime checks for the issue, though.

Reply:
So we leave it for now?

Reply:
I also noticed that I would need to include the unix_domain_socket.hpp in iceoryx_posh_types.hpp so I would rather leave it for now.